### ISSUE TITLE
Running sqlfluff lint raises ModuleNotFoundError: No module named 'dbt.exceptions'

### ISSUE THREAD

**Issue by robertdefilippi** [sqlfluff/sqlfluff#965]
## Expected Behaviour

When I run `sqlfluff lint` on a file or folder, I would expect the issues with my file to be sent back to me.

## Observed Behaviour

When I run `$ sqlfluff lint . ` or  `$ sqlfluff lint path/to/file.sql`  or  `$ sqlfluff lint path/to/folder` I get the error message below:

```
WARNING    Unable to lint models/marts/trends/mdl/this_file.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/Users/me/.pyenv/versions/3.8.3/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1412, in lint_path
    self.lint_string(
  File "/Users/me/.pyenv/versions/3.8.3/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 1194, in lint_string
    parsed = self.parse_string(in_str=in_str, fname=fname, config=config)
  File "/Users/me/.pyenv/versions/3.8.3/lib/python3.8/site-packages/sqlfluff/core/linter.py", line 874, in parse_string
    templated_file, templater_violations = self.templater.process(
  File "/Users/me/.pyenv/versions/3.8.3/lib/python3.8/site-packages/sqlfluff/core/templaters/dbt.py", line 196, in process
    from dbt.exceptions import (
ModuleNotFoundError: No module named 'dbt.exceptions'
```

> this_file.sql
```
{{
    config(
        materialized='insert_by_period',
        period='month',
        timestamp_field='t_datetime',
        start_date='2017-01-01',
        sort=['t_datetime', 't_id'],
        dist='tag_id',
    )
}}

SELECT
    -- primary_key
    ROW_NUMBER() OVER (PARTITION BY 1)                  AS row_id,

    -- foreign_keys
    g_id::VARCHAR                                   AS s_id,
    t_id::VARCHAR                                    AS t_id,

    -- fields
    createdate::TIMESTAMP                               AS st_datetime,

    -- meta_data
    SYSDATE                                             AS _added_at

FROM
    {{ source('pub', 'gt') }}

WHERE
  __PERIOD_FILTER__

ORDER BY
  row_id
```

However I get this message with _all_ files and folders, so I don't know if it a problem with the way I've written all my files or with sqlfluff. I've ignored and unignored the files to see if that was the issue, but I get the error either way.

## Version

❯ sqlfluff version 
0.5.2

❯ python --version
Python 3.8.3

❯ dbt --version
installed version: 0.19.0

## Configuration

> .sqlfluff
```
# SQLfluff rules:
# https://docs.sqlfluff.com/en/latest/rules.html

[sqlfluff]
verbose = 1
dialect = postgres
nocolor = False
exclude_rules = L009,L014,L039
ignore_templated_areas = True
templater = dbt
recurse = 0
output_line_length = 80
runaway_limit = 10
ignore = parsing

[sqlfluff:indentation]
indented_joins = False
template_blocks_indent = True

[sqlfluff:rules]
tab_space_size = 2
max_line_length = 120
indent_unit = space

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L016]
ignore_comment_lines = True

[sqlfluff:rules:L019]
comma_style = trailing

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = join
```

> .sqlfluffignore
```
/admin/
/analysis/
/data/
/dbt_modules/
/macros/
/models/
!/models/staging/public/*.sql
/logs/
/snapshots/
/target/
/tests/
```

**Comment by alanmcruickshank** [comment]
This is very strange. It implies that sqlfluff can't find dbt 🤔 .

Rather than running `dbt --version`, can you run `pip list` to see what versions of `sqlfluff` and `dbt` it finds? If my theory is right then while you can call `dbt` as a command, you won't find it in `pip list`.

If that's the case then you might not have dbt installed in your current virtual environment. If that is the case I wonder if we need a more helpful error message here 🤔 .

**Comment by robertdefilippi** [comment]
So I don't have dbt installed using `pip` I have it installed using homebrew. Is there a way to point sqlfluff to that version ... or must I have it installed with pip?

```
❯ pip list | grep sqlfluff
sqlfluff                        0.5.2

❯ pip list | grep dbt

❯ which dbt
/usr/local/bin/dbt
```

**Comment by NiallRees** [comment]
@robertdefilippi dbt must be installed with pip.
